benchmark_stream_test.go

```diff
@@ -0,0 +1,255 @@
+		},
+		{
+			name:                                 "server stream success",
+			procedure:                            "Bar::ServerStream",
+			method:                               "Bar/ServerStream",
+			numConnections:                       10,
+			requests:                             [][]byte{nil},
+			returnOutput:                         []simple.Foo{{}},
+			expectedInput:                        []simple.Foo{{}},
+			expectedStreamsOpened:                10,
+			expectedServerSentStreamMessages:     10,
+			expectedServerReceivedStreamMessages: 10,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svc := &simpleService{
+				expectedInput: tt.expectedInput,
+				returnOutput:  tt.returnOutput,
+			}
+			lis, server := setupGRPCServer(t, svc)
+			defer server.Stop()
+
+			req := &transport.Request{
+				TargetService: "foo",
+				Method:        tt.procedure,
+				Timeout:       time.Second,
+			}
+
+			source, err := protobuf.NewDescriptorProviderFileDescriptorSetBins("./testdata/protobuf/simple/simple.proto.bin")
+			require.NoError(t, err)
+
+			serializer, err := encoding.NewProtobuf(tt.method, source)
+			require.NoError(t, err)
+
+			bench := benchmarkStreamMethod{
+				serializer:            serializer,
+				streamRequest:         &transport.StreamRequest{Request: req},
+				streamRequestMessages: tt.requests,
+			}
+
+			transports, err := warmTransports(bench, tt.numConnections, TransportOptions{
+				ServiceName: "foo",
+				CallerName:  "test",
+				Peers:       []string{"grpc://" + lis.String()},
+			}, _resolvedGrpcProto, 1)
+			require.NoError(t, err)
+
+			for i, transport := range transports {
+				assert.NotNil(t, transport, "transports[%d] must not be nil", i)
+			}
+
+			assert.Equal(t, tt.expectedStreamsOpened, svc.streamsOpened.Load())
+			assert.Equal(t, tt.expectedServerSentStreamMessages, svc.serverSentStreamMessages.Load())
+			assert.Equal(t, tt.expectedServerReceivedStreamMessages, svc.serverReceivedStreamMessages.Load())
+		})
+	}
+}
+
+func TestStreamBenchmarkCallMethod(t *testing.T) {
+	tests := []struct {
+		name                                 string
+		procedure                            string
+		method                               string
+		requests                             [][]byte
+		returnOutput                         []simple.Foo
+		expectedInput                        []simple.Foo
+		expectedStreamsOpened                int32
+		expectedServerSentStreamMessages     int32
+		expectedServerReceivedStreamMessages int32
+	}{
+		{
+			name:                                 "client stream success",
+			procedure:                            "Bar::ClientStream",
+			method:                               "Bar/ClientStream",
+			requests:                             [][]byte{nil, nil},
+			returnOutput:                         []simple.Foo{{}},
+			expectedInput:                        []simple.Foo{{}, {}},
+			expectedStreamsOpened:                1,
```

Comment:
same here. a little comment after those numbers will be really helpful for future maintenance purpose.